src/components/Topology/Topology.js

```diff
@@ -205,24 +282,27 @@ export default ({ modelData, width, height }) => {
       .attr("stroke-width", 3)
       .attr("stroke", "#888888")
       .call((_) => {
-        // When ever a new element is added zoom the canvas to fit.
-        const {
-          width: svgWidth,
-          height: svgHeight,
-        } = topo.node().getBoundingClientRect();
+        const topoNode = topo.node();
+        const { width: svgWidth, height: svgHeight } = topoNode
+          ? topoNode.getBoundingClientRect()
+          : {};
+
+        // Whenever a new element is added zoom the canvas to fit.
         if (svgWidth > 0 && svgHeight > 0) {
-          // Magic number that presents reasonable padding around the viz.
-          const padding = 200;
-          const scale = Math.min(
-            width / (svgWidth + padding),
-            height / (svgHeight + padding)
-          );
-          const translateX = (width - svgWidth * scale) / 2;
-          const translateY = (height - svgHeight * scale) / 2;
-          topo.attr(
-            "transform",
-            `translate(${translateX},${translateY}) scale(${scale},${scale})`
-          );
+          const scale = Math.min(width / svgWidth, height / svgHeight) || 1;
```

Comment:
I'm not entirely sure where the problem is happening but it doesn't appear to be able to handle negative values properly. Note how in the following screenshot the icons are spilling out of the Topology bounadry. This works properly in the production release.

![Screen Shot 2020-06-15 at 11 33 11 AM](https://user-images.githubusercontent.com/532033/84689903-eaa79980-aefe-11ea-9c12-522d0826863f.png)

Reply:
https://github.com/canonical-web-and-design/jaas-dashboard/issues/559